### PR TITLE
Add nplusone and pytest-pythonpath Python packages

### PR DESCRIPTION
**Pull request by cript0nauta**
###### Motivation for this change

Add the `nplusone` Python package, which is a dependency of [Faraday](github.com/infobyte/faraday/).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/nplusone/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    owner = "jmcarp";
+    repo = "nplusone";
+    rev = "v${version}";
+    sha256 = "0qdwpvvg7dzmksz3vqkvb27n52lq5sa8i06m7idnj5xk2dgjkdxg";
+  };
+
+  # The tests assume the source code is in an nplusone/ directory. When using
+  # the Nix sandbox, it will be in a source/ directory instead, making the
+  # tests fail.
+  patches = [ ./fix_tests.diff ];
```
Comment (jonringer) [inline pkgs/development/python-modules/nplusone/default.nix:20]: I would prefer substituteInPlace over a diff, generally they are less likely to fail to apply when bumping the package.
```suggestion
  prePatch = ''
    substituteInPlace tests/conftest.py \
      --replace nplusone/tests/conftest source/tests/conftest
  '';
```

File: pkgs/development/python-modules/pytest-pythonpath/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    inherit pname version;
+    sha256 = "0qhxh0z2b3p52v3i0za9mrmjnb1nlvvyi2g23rf88b3xrrm59z33";
+  };
+
+  checkPhase = ''
+    pytest
+  '';
+
+  checkInputs = [ pytest ];
+  propagatedBuildInputs = [ pytest ];
```
Comment (jonringer) [inline pkgs/development/python-modules/pytest-pythonpath/default.nix:17]: typical ordering:
```suggestion
  propagatedBuildInputs = [ pytest ];
  checkInputs = [ pytest ];

  checkPhase = ''
    pytest
  '';
```

File: pkgs/development/python-modules/nplusone/default.nix (hunk 1 above)
Comment (cript0nauta) [reply pkgs/development/python-modules/nplusone/default.nix]: Done, nice suggestion!

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, the commit message should be:
```
pythonPackages.pytest-pythonpath: init at 0.7.3
pythonPackages.nplusone: init at 1.0.0 
```

you can use `git rebase -i` to re-order and squash the commits in the correct order

**Comment by cript0nauta** [comment]
I didn't fully undertand the suggestion about the commit messages. Right now, all commits are using the `<pkg-name>: <subject-line>` convention:

```
$ git log master..nplusone --reverse --oneline
95027da5267 pythonPackages.pytest-pythonpath: init at 0.7.3
9128e87de66 pythonPackages.nplusone: init at 1.0.0
5dfa72af62d pythonPackages.pytest-pythonpath: cosmetic changes
a5ac33d7b13 pythonPackages.nplusone: don't use patches
```

Did you mean I should squash 5dfa72af62d and a5ac33d7b13 into the previous commits so we only have the `init at ...` commits?

**Comment by jonringer** [comment]
I meant that the fixup commits should be squashed into their respective package addition commits

**Comment by cript0nauta** [comment]
> I meant that the fixup commits should be squashed into their respective package addition commits

Done, thanks for clarifying!

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 94054 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>5 packages built:</summary><br>- python27Packages.pytest-pythonpath<br>- python37Packages.nplusone<br>- python37Packages.pytest-pythonpath<br>- python38Packages.nplusone<br>- python38Packages.pytest-pythonpath</details>

**Pull request merged**